### ISSUE TITLE
[BUG] Cannot import from egg

### ISSUE THREAD

**Issue by rayzchen** [pyunity/pyunity#43]
**Describe the bug**

Trying to import PyUnity from an egg file errors out.

**To Reproduce**

Run `cython=0 python setup.py bdist_egg`
Go to `dist` folder
Run code

**Expected behavior**

Successful import

**Code**
```python3
import sys
sys.path.insert(0, "pyunity-0.9.0-py3.9.egg")
import pyunity
```

**System info**
```
##############################
         VERSION INFO
##############################
PyUnity version: v0.9.0
Python version: v3.9.4-final
Operating system: Windows 10
Architecture: 32bit
Requirements:
- pyopengl version: 3.1.5
- pillow version: 9.0.1
- pysdl2 version: 0.9.11
- pysdl2-dll version: 2.0.20
- pyglm version: 2.5.7
- importlib-resources version: None
- glfw version: 2.5.0
```

**Traceback**
```
Traceback (most recent call last):
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\importlib\_common.py", line 35, in _tempfile
    os.write(fd, reader())
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\zipfile.py", line 2322, in read_bytes
    with self.open('rb') as strm:
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\zipfile.py", line 2306, in open
    stream = self.root.open(self.at, zip_mode, pwd=pwd)
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\zipfile.py", line 1502, in open
    zinfo = self.getinfo(name)
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\zipfile.py", line 1429, in getinfo
    raise KeyError(
KeyError: "There is no item named 'pyunity/shaders/skybox/textures/' in the archive"
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "pyunity-0.9.0-py3.9.egg\pyunity\__init__.py", line 209, in <module>
    from .render import __all__ as _render_all
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "pyunity-0.9.0-py3.9.egg\pyunity\render.py", line 324, in <module>
    skyboxes["Water"] = Skybox(stack.enter_context(as_file(ref / "skybox/textures")))
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\contextlib.py", line 429, in enter_context
    result = _cm_type.__enter__(cm)
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\importlib\_common.py", line 52, in as_file
    with _tempfile(path.read_bytes, suffix=path.name) as local:
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "C:\Users\raych\AppData\Local\Programs\Python\Python39-32\lib\importlib\_common.py", line 40, in _tempfile
    os.remove(raw_path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\raych\\AppData\\Local\\Temp\\tmpbrwljrmttextures'
```

**Additional context**

**Comment by rayzchen** [comment]
The problem with this is that `Skybox.fromFolder` requires all 6 images in one folder and `importlib.resources` can only load one file at a time into an unnamed tempfile.

**Comment by rayzchen** [comment]
One possibility is to extract the resources into a folder in `Path.home()` like Kivy does. This could be maintained with a separate module that is specific to managing resources, especially in PyUnity projects. (potential name: FileLoader?)

**Comment by rayzchen** [comment]
Will be fixed by #63
